tc-messaging/src/main/java/com/tc/bytes/TCByteBufferAllocator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ *  The contents of this file are subject to the Terracotta Public License Version
+ *  2.0 (the "License"); You may not use this file except in compliance with the
+ *  License. You may obtain a copy of the License at
+ *
+ *  http://terracotta.org/legal/terracotta-public-license.
+ *
+ *  Software distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for
+ *  the specific language governing rights and limitations under the License.
+ *
+ *  The Covered Software is Terracotta Core.
+ *
+ *  The Initial Developer of the Covered Software is
+ *  Terracotta, Inc., a Software AG company
+ *
+ */
+package com.tc.bytes;
+
+import com.tc.util.concurrent.SetOnceFlag;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.function.Supplier;
+
+/**
+ *
+ */
+public class TCByteBufferAllocator {
+  private final Supplier<TCByteBuffer> newBuffers;
+  private final Queue<TCByteBuffer> returns;
+  private final Collection<TCByteBuffer> items = new LinkedList<>();
```

Comment:
This obviously leans heavily on the ordering right, should probably type it as `List` just to be safe. Also why a `LinkedList`? `ArrayList` is probably a better choice, no?

Reply:
ArrayList is better.